lib/ansible/modules/cloud/docker/docker_network.py

```diff
@@ -208,14 +208,12 @@ def __init__(self, client):
             self.absent()
 
     def get_existing_network(self):
-        networks = self.client.networks(names=[self.parameters.network_name])
-        # check if a user is trying to find network by its Id
-        if not networks:
-            networks = self.client.networks(ids=[self.parameters.network_name])
-        if not networks:
-            return None
-        else:
-            return networks[0]
+        networks = self.client.networks()
+        network = None
+        for n in networks:
+            if n['Name'] == self.parameters.network_name or n['Id'] == self.parameters.network_name:
+                network = n
```

Comment:
`network` variable is extraneous, simply `return n` (`network` could be used instead of `n`) when the condition is true:
```python
def get_existing_network(self):
    for network in self.client.networks():
        if n['Name'] == self.parameters.network_name or n['Id'] == self.parameters.network_name:
            return network
```